lib/internal/worker.js

```diff
@@ -70,8 +70,14 @@ class Worker extends EventEmitter {
                                  'string', filename);
     }
 
-    if (!options.eval && !path.isAbsolute(filename)) {
-      throw new errors.TypeError('ERR_WORKER_NEED_ABSOLUTE_PATH', filename);
+    if (!options.eval) {
+      if (!path.isAbsolute(filename)) {
+        throw new errors.TypeError('ERR_WORKER_NEED_ABSOLUTE_PATH', filename);
+      }
+      const ext = path.extname(filename);
+      if (ext !== '.js') {
+        throw new errors.TypeError('ERR_WORKER_UNSUPPORTED_EXTENSION', ext);
+      }
```

Comment:
Should we also allow `.mjs` when the `--experimental-modules` flag is used?

Reply:
https://github.com/ayojs/ayo/pull/58#issuecomment-330986097

Reply:
Ah, missed the recent comments.

Reply:
That's on the roadmap, but hasn't been implemented yet.